cmd/coder/urls.go

```diff
@@ -33,55 +55,110 @@ var urlAccessLevel = map[string]string{
 	"PUBLIC":  "Anyone on the internet can access this link",
 }
 
-func portIsValid(port string) bool {
+func validatePort(port string) (int, error) {
 	p, err := strconv.ParseUint(port, 10, 16)
+	if err != nil {
+		flog.Error("Invalid port")
+		return 0, err
+	}
 	if p < 1 {
 		// port 0 means 'any free port', which we don't support
 		err = strconv.ErrRange
+		flog.Error("Port must be > 0")
+		return 0, err
 	}
-	if err != nil {
-		fmt.Println("Invalid port")
-	}
-	return err == nil
+	return int(p), nil
 }
 
 func accessLevelIsValid(level string) bool {
 	_, ok := urlAccessLevel[level]
 	if !ok {
-		fmt.Println("Invalid access level")
+		flog.Error("Invalid access level")
 	}
 	return ok
 }
 
+type listSubCmd struct {
+	outputFmt string
+}
+
+// Run gets the list of active devURLs from the cemanager for the
+// specified environment and outputs info to stdout.
+func (sub listSubCmd) Run(fl *pflag.FlagSet) {
+	envName := fl.Arg(0)
```

Comment:
Validate that this is nonempty before continuing.

Reply:
OK. Currently it'll output `[]` for json and nothing for tabbed. You want no output for both or a msg such as `No devurls defined` for human output?

Reply:
Changed for now to output nothing for empty devURLs